test/test_collector.py

```diff
@@ -1186,6 +1187,72 @@ def weight_reset(m):
         m.reset_parameters()
 
 
+class TestPreemptiveThreshold:
+    @pytest.mark.parametrize("env_name", ["conv", "vec"])
+    def test_sync_collector_interruptor_mechanism(self, env_name, seed=100):
+        def env_fn(seed):
+            env = make_make_env(env_name)()
+            env.set_seed(seed)
+            return env
+
+        policy = make_policy(env_name)
+        interruptor = Interruptor()
+        interruptor.start_collection()
+
+        collector = SyncDataCollector(
+            create_env_fn=env_fn,
+            create_env_kwargs={"seed": seed},
+            policy=policy,
+            frames_per_batch=50,
+            total_frames=200,
+            device="cpu",
+            interruptor=interruptor,
+            split_trajs=False,
+        )
+
+        interruptor.stop_collection()
+        for batch in collector:
+            assert batch["collector"]["traj_ids"][0] != -1
+            assert batch["collector"]["traj_ids"][1] == -1
+
+    @pytest.mark.parametrize("env_name", ["conv", "vec"])
+    def test_multisync_collector_interruptor_mechanism(self, env_name, seed=100):
+        def env_fn(seed):
+            env = make_make_env(env_name)()
+            env.set_seed(seed)
+            return env
+
+        policy = make_policy(env_name)
+
+        collector = MultiSyncDataCollector(
+            create_env_fn=[env_fn, env_fn, env_fn, env_fn],
+            create_env_kwargs=[
+                {"seed": seed},
+                {"seed": seed},
+                {"seed": seed},
+                {"seed": seed},
+            ],
+            policy=policy,
+            total_frames=800,
+            max_frames_per_traj=50,
+            frames_per_batch=800,
+            init_random_frames=-1,
+            reset_at_each_iter=False,
+            devices="cpu",
+            storing_devices="cpu",
+            split_trajs=False,
+            preemptive_threshold=0.25,
+        )
+
+        for batch in collector:
+            assert (
+                batch["collector"]["traj_ids"][
```

Comment:
let's rewrite this in a more straightforward way, it's a bit hard to read
What if none is -1? Shouldn't we test that we have at least one traj_id set to -1 in the batch?

Reply:
I simplified the code for more clarity.

Regarding the possibility of having none -1, I have addressed that as well by setting the preemptive threshold to 0.0. This way, all collectors will stop after the first iteration and only the very first frame of each sync collector will be valid. So we know for sure there are -1's in the batch.